packages/babel-plugin/src/class-names/index.tsx

```diff
@@ -108,21 +168,36 @@ export const visitClassNamesPath = (path: NodePath<t.JSXElement>, meta: Metadata
 
   // Second pass to replace all usages of `style`.
   path.traverse({
-    Identifier(path) {
-      if (
-        path.node.name !== 'style' ||
-        path.parentPath.isProperty() ||
-        !path.scope.hasOwnBinding('style')
-      ) {
-        // Nothing to do - skip.
-        return;
+    Expression(path) {
```

Comment:
We need to traverse `style` only after `css` traversal is done otherwise it's breaking in case when `style` is not `undefined`.

Check test `should replace style identifier with css variable object` from `packages/babel-plugin/src/class-names/__tests__/behaviour.test.tsx` file.